src/transformers/utils/import_utils.py

```diff
@@ -288,30 +288,42 @@ def is_torch_bf16_available():
     # 4. torch.autocast exists
     # XXX: one problem here is that it may give invalid results on mixed gpus setup, so it's
     # really only correct for the 0th gpu (or currently set default device if different from 0)
-    is_torch_gpu_bf16_available = True
-    is_torch_cpu_bf16_available = True
     if version.parse(torch.__version__) < version.parse("1.10"):
-        is_torch_gpu_bf16_available = False
-        is_torch_cpu_bf16_available = False
+        return False
 
     if torch.cuda.is_available() and torch.version.cuda is not None:
         if torch.cuda.get_device_properties(torch.cuda.current_device()).major < 8:
-            is_torch_gpu_bf16_available = False
+            return False
         if int(torch.version.cuda.split(".")[0]) < 11:
-            is_torch_gpu_bf16_available = False
+            return False
         if not hasattr(torch.cuda.amp, "autocast"):
-            is_torch_gpu_bf16_available = False
+            return False
     else:
-        is_torch_gpu_bf16_available = False
+        return False
+
+    return True
+
+
+def is_torch_bf16_cpu_available():
+    if not is_torch_available():
+        return False
+
+    import torch
+
+    if version.parse(torch.__version__) < version.parse("1.10"):
+        return False
 
-    # checking CPU
     try:
         # multiple levels of AttributeError depending on the pytorch version so do them all in one check
         _ = torch.cpu.amp.autocast
     except AttributeError:
-        is_torch_cpu_bf16_available = False
+        return False
+
+    return True
 
-    return is_torch_cpu_bf16_available or is_torch_gpu_bf16_available
+
+def is_torch_bf16_available():
+    return is_torch_bf16_cpu_available() or is_torch_bf16_gpu_available()
```

Comment:
Now that you have split it up into 2 specific components - I think this is ambiguous - what does it actually mean from the usage point of view? 

say, a user has a cpu supporting bf16, but they are actually planning to use a gpu, so this will return True and then their code will either fail or run really slow.

I know that you haven't added this in this PR, I missed that expansion to add cpu checks in the previous PR.

What do you think?

The original ` is_torch_bf16_available` was just doing gpu checks, so perhaps we deprecate it and alias to `is_torch_bf16_gpu_available`

Reply:
```
def is_torch_bf16_available(...):
    warn(deprecated)
    return is_torch_bf16_gpu_available(...)
```

Reply:
that's correct, we rename it in the Trainer, as it's incorrect - it should be doing a cpu check as introduced in this PR

Reply:
can do in the next PR